controllers/doguExportManager.go

```diff
@@ -0,0 +1,147 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudogu/k8s-dogu-operator/v3/api/ecoSystem"
+	"github.com/cloudogu/k8s-dogu-operator/v3/controllers/resource"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"time"
+
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	k8sv2 "github.com/cloudogu/k8s-dogu-operator/v3/api/v2"
+)
+
+const (
+	ChangeExportModeEventReason        = "ChangeExportMode"
+	ErrorOnChangeExportModeEventReason = "ErrChangeExportMode"
+)
+
+type exportModeNotYetChangedError struct {
+	doguName               string
+	desiredExportModeState bool
+}
+
+func (e exportModeNotYetChangedError) Error() string {
+	return fmt.Sprintf("the export-mode of dogu %q has not yet been changed to its desired state: %v", e.doguName, e.desiredExportModeState)
+}
+
+func (e exportModeNotYetChangedError) Requeue() bool {
+	return true
+}
+
+func (e exportModeNotYetChangedError) GetRequeueTime() time.Duration {
+	return requeueWaitTimeout
+}
+
+type doguExportManager struct {
+	doguClient       ecoSystem.DoguInterface
+	podClient        podInterface
+	resourceUpserter resource.ResourceUpserter
+	doguFetcher      localDoguFetcher
+	eventRecorder    record.EventRecorder
+}
+
+func NewDoguExportManager(
+	doguClient ecoSystem.DoguInterface,
+	podClient podInterface,
+	resourceUpserter resource.ResourceUpserter,
+	doguFetcher localDoguFetcher,
+	eventRecorder record.EventRecorder,
+) *doguExportManager {
+	return &doguExportManager{
+		doguClient:       doguClient,
+		podClient:        podClient,
+		resourceUpserter: resourceUpserter,
+		doguFetcher:      doguFetcher,
+		eventRecorder:    eventRecorder,
+	}
+}
+
+func (dem *doguExportManager) shouldUpdateExportMode(ctx context.Context, doguResource *k8sv2.Dogu) bool {
+	logger := log.FromContext(ctx)
+
+	shouldExportModeBeActive := doguResource.Spec.ExportMode
+
+	isExportModeActive, err := dem.isDeploymentInExportMode(ctx, doguResource.GetObjectKey())
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("failed to check if deployment is in export-mode dogu %q", doguResource.Name))
+		return true
```

Comment:
Why is the error ignore-logged but the code one level higher manages to return errors? Shouldn't this function return the error here as well?